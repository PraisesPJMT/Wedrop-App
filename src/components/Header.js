import React, { useState } from 'react';
import { Link, NavLink, useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faCloudSunRain, faMagnifyingGlassLocation, faBars, faClose,
} from '@fortawesome/free-solid-svg-icons';
import { allStatus, allWeather, getWeather } from '../redux/weather/weather';
import api from '../redux/api/api';

const Header = () => {
  const [menuOpen, setMenu] = useState(false);
  const status = useSelector(allStatus);
  const weather = useSelector(allWeather);
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const navs = [
    { name: 'Home', id: 1, path: '/' },
    { name: 'Contact', id: 2, path: '/contact' },
  ];

  const handleMobileMenu = () => {
    setMenu((prevState) => !prevState);
  };

  const closeMobileMenu = () => {
    setMenu(() => false);
  };

  const navigateToCity = () => {
    closeMobileMenu();
    const search = document.querySelector('input[name="search"]');
    if (search.value.length > 2) {
      dispatch(getWeather(search.value));
      if (status === 'succeeded') {
        const { ContinentName, CountryName } = api.getContinentCountryName(weather.countryCode);
        navigate(`/${ContinentName.toLowerCase().replace(/ /g, '-').replace(/'/g, '')}/${CountryName.toLowerCase().replace(/ /g, '-')}/${weather.cityName.toLowerCase().replace(/ /g, '-')}`);
        search.value = '';
      }
    }
  };

  return (
    <header className="py-2 px-4 bg-steel-blue flex item-center justify-between z-[100]">
      <div className="w-[20px] flex items-center md:hidden z-[100]">
        <button
          className="flex item-center"
          type="button"
          onClick={handleMobileMenu}
        >
          <FontAwesomeIcon
            id="mobileMenu"
            className="text-xl text-white"
            icon={menuOpen ? faClose : faBars}
          />
        </button>
      </div>
      <Link className="flex items-center gap-1 text-white hover:scale-110 z-[100]" to="/" onClick={closeMobileMenu}>
        <FontAwesomeIcon
          className="text-xl md:text-2xl"
          icon={faCloudSunRain}
        />
        <span className="capitalize text-xm md:text-xl font-bold">wedrop</span>
      </Link>
      <nav className={`pb-6 flex flex-col gap-6 bg-steel-blue z-[10] absolute left-0 w-full h-screen transition-all duration-500 ease-in-out md:order-last md:flex-row md:items-center md:pb-0 md:static md:z-auto md:w-auto md:h-auto md:pt-0 md:pl-9 ${menuOpen ? 'top-0 pt-16' : 'top-[-890px] pt-auto'}`}>
        { navs.map((nav) => (
          <NavLink
            className="text-white font-bold text-xl px-6 w-full md:w-auto md:px-1.5 md:mx-1 active:border-b-2 active:border-b-white md:hover:scale-110"
            key={nav.path}
            to={nav.path}
            onClick={closeMobileMenu}
          >
            {nav.name}
          </NavLink>
        ))}
      </nav>
      <div className="input-group relative flex w-2/5 flex-wrap items-stretch shadow-md z-[100]">
        <input
          type="search"
          name="search"
          className="form-control relative flex-auto w-4/5 block px-1 py-0.1 text-steel-blue text-bold text-xs font-normal bg-white bg-clip-padding border-white border-2 rounded-l-full transition ease-in-out focus:text-steel-blue focus:bg-white focus:border-white focus:outline-none md:w-auto md:text-xl md:focus:text-xl md:pl-2.5"
          placeholder="Search City..."
          aria-label="Search"
          aria-describedby="button-addon2"
          onKeyPress={(event) => {
            if (event.key === 'Enter') navigateToCity();
          }}
        />
        <button
          className="btn inline-block bg-steel-blue w-1/5 md:w-auto text-white font-medium text-xs leading-tight uppercase rounded-r shadow-md border-white border-2 hover:shadow-lg focus:bg-steel-blue  focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg active:text-steel-blue active:text-bold transition duration-150 ease-in-out flex items-center justify-center md:px-2 md:py-0.5"
          type="button"
          id="button-addon2"
          onClick={() => navigateToCity()}
        >
          <FontAwesomeIcon
            className="text-xs hover:scale-125 px-1.5 py-1.5 md:text-xl md:p-auto"
            icon={faMagnifyingGlassLocation}
          />
        </button>
      </div>
    </header>
  );
};

export default Header;
