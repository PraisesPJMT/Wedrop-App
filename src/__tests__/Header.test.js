import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Header from '../components/Header';

describe('Header component tests', () => {
  test('Renders Header component correctly', () => {
    const header = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );
    expect(header).toMatchSnapshot();
  });

  test('Check for heading in the header \'Wedrop\' is in the document', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );
    expect(
      screen.getByText(/Wedrop/i),
    ).toBeInTheDocument();
  });

  test('Check for City search placeholder is in the document', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );
    expect(
      screen.getByPlaceholderText('Search City...'),
    ).toBeInTheDocument();
  });

  test('Check for link in the Header \'Home\' is in the document', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );
    expect(
      screen.getByRole('link', { name: /Home/i }),
    ).toBeInTheDocument();
  });

  test('Check for link in the Header \'Contact\' is in the document', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );
    expect(
      screen.getByRole('link', { name: /Contact/i }),
    ).toBeInTheDocument();
  });
});
