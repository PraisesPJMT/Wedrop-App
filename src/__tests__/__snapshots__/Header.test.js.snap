// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests Renders Header component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="py-2 px-4 bg-steel-blue flex item-center justify-between z-[100]"
      >
        <div
          class="w-[20px] flex items-center md:hidden z-[100]"
        >
          <button
            class="flex item-center"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-chevron-left text-xl text-white"
              data-icon="chevron-left"
              data-prefix="fas"
              focusable="false"
              id="mobileMenu"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 278.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <a
          class="flex items-center gap-1 text-white hover:scale-110 z-[100]"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cloud-sun-rain text-xl md:text-2xl"
            data-icon="cloud-sun-rain"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M137.9 3c-4.5-3.2-10.3-3.9-15.4-1.8s-8.8 6.7-9.7 12.2L98.7 98.7 13.4 112.8c-5.5 .9-10.1 4.6-12.2 9.7S-.2 133.4 3 137.9l50.3 70.3L3 278.5c-3.2 4.5-3.9 10.3-1.8 15.4s6.7 8.8 12.2 9.7l85.3 14.1L112.8 403c.9 5.5 4.6 10.1 9.7 12.2s10.9 1.4 15.4-1.8l55.2-39.5c-19.9-21.9-32.3-50.8-33.1-82.6c-17.6-10.2-32.2-26.1-40.6-46.3c-20.3-49 3-105.1 52-125.4c29.4-12.2 61.4-8.7 86.7 6.7c13.1-21.8 32.5-39.4 55.8-50.3L303.6 13.4c-.9-5.5-4.6-10.1-9.7-12.2S282.9-.2 278.5 3L208.2 53.3 137.9 3zM291.4 415.9c-2.5 .6-5.1 .6-7.6-.1c1.4 0 2.8 .1 4.2 .1h3.5zm-46.3-260c-17.3-12.2-40.4-15.6-61.5-6.9c-32.7 13.5-48.2 51-34.6 83.6c3.6 8.7 8.9 16.1 15.3 22.2c10.6-39.7 39.9-71.8 77.8-86.4c.8-4.3 1.8-8.4 3-12.5zM277.4 420c-11-7.4-25.9-4.4-33.3 6.7l-32 48c-7.4 11-4.4 25.9 6.7 33.3s25.9 4.4 33.3-6.7l32-48c7.4-11 4.4-25.9-6.7-33.3zm96 0c-11-7.4-25.9-4.4-33.3 6.7l-32 48c-7.4 11-4.4 25.9 6.7 33.3s25.9 4.4 33.3-6.7l32-48c7.4-11 4.4-25.9-6.7-33.3zm96 0c-11-7.4-25.9-4.4-33.3 6.7l-32 48c-7.4 11-4.4 25.9 6.7 33.3s25.9 4.4 33.3-6.7l32-48c7.4-11 4.4-25.9-6.7-33.3zm96 0c-11-7.4-25.9-4.4-33.3 6.7l-32 48c-7.4 11-4.4 25.9 6.7 33.3s25.9 4.4 33.3-6.7l32-48c7.4-11 4.4-25.9-6.7-33.3zm74.5-116.1c0-39.3-28.4-72.1-65.8-78.7c1.2-5.6 1.9-11.3 1.9-17.2c0-44.2-35.8-80-80-80c-17 0-32.8 5.3-45.8 14.4C433.3 114.6 402.8 96 368 96c-53 0-96 43-96 96l0 1.3c-45.4 7.6-80 47.1-80 94.6c0 53 43 96 96 96H559.9c44.2 0 80-35.8 80-80z"
              fill="currentColor"
            />
          </svg>
          <span
            class="capitalize text-xm md:text-xl font-bold"
          >
            wedrop
          </span>
        </a>
        <nav
          class="pb-6 flex flex-col gap-6 bg-steel-blue z-[10] absolute left-0 w-full h-screen transition-all duration-500 ease-in-out md:order-last md:flex-row md:items-center md:pb-0 md:static md:z-auto md:w-auto md:h-auto md:pt-0 md:pl-9 top-[-890px] pt-auto"
        >
          <a
            aria-current="page"
            class="text-white font-bold text-xl px-6 w-full md:w-auto md:px-1.5 md:mx-1 active:border-b-2 active:border-b-white md:hover:scale-110 active"
            href="/"
          >
            Home
          </a>
          <a
            class="text-white font-bold text-xl px-6 w-full md:w-auto md:px-1.5 md:mx-1 active:border-b-2 active:border-b-white md:hover:scale-110"
            href="/contact"
          >
            Contact
          </a>
        </nav>
        <div
          class="input-group relative flex w-2/5 flex-wrap items-stretch shadow-md z-[100]"
        >
          <input
            aria-describedby="button-addon2"
            aria-label="Search"
            class="form-control relative flex-auto w-4/5 block px-1 py-0.1 text-steel-blue text-bold text-xs font-normal bg-white bg-clip-padding border-white border-2 rounded-l-full transition ease-in-out focus:text-steel-blue focus:bg-white focus:border-white focus:outline-none md:w-auto md:text-xl md:focus:text-xl md:pl-2.5"
            name="search"
            placeholder="Search City..."
            type="search"
          />
          <button
            class="btn inline-block bg-steel-blue w-1/5 md:w-auto text-white font-medium text-xs leading-tight uppercase rounded-r shadow-md border-white border-2 hover:shadow-lg focus:bg-steel-blue  focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg active:text-steel-blue active:text-bold transition duration-150 ease-in-out flex items-center justify-center md:px-2 md:py-0.5"
            id="button-addon2"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-magnifying-glass-location text-xs hover:scale-125 px-1.5 py-1.5 md:text-xl md:p-auto"
              data-icon="magnifying-glass-location"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM288 176c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 48.8 46.5 111.6 68.6 138.6c6 7.3 16.8 7.3 22.7 0c22.1-27 68.6-89.8 68.6-138.6zm-48 0c0 17.7-14.3 32-32 32s-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="w-[20px] flex items-center md:hidden z-[100]"
        >
          <button
            class="flex item-center"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-bars text-xl text-white"
              data-icon="bars"
              data-prefix="fas"
              focusable="false"
              id="mobileMenu"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="py-2 px-4 bg-steel-blue flex item-center justify-between z-[100]"
    >
      <div
        class="w-[20px] flex items-center md:hidden z-[100]"
      >
        <button
          class="flex item-center"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-left text-xl text-white"
            data-icon="chevron-left"
            data-prefix="fas"
            focusable="false"
            id="mobileMenu"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l192 192c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 278.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-192 192z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <a
        class="flex items-center gap-1 text-white hover:scale-110 z-[100]"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-cloud-sun-rain text-xl md:text-2xl"
          data-icon="cloud-sun-rain"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M137.9 3c-4.5-3.2-10.3-3.9-15.4-1.8s-8.8 6.7-9.7 12.2L98.7 98.7 13.4 112.8c-5.5 .9-10.1 4.6-12.2 9.7S-.2 133.4 3 137.9l50.3 70.3L3 278.5c-3.2 4.5-3.9 10.3-1.8 15.4s6.7 8.8 12.2 9.7l85.3 14.1L112.8 403c.9 5.5 4.6 10.1 9.7 12.2s10.9 1.4 15.4-1.8l55.2-39.5c-19.9-21.9-32.3-50.8-33.1-82.6c-17.6-10.2-32.2-26.1-40.6-46.3c-20.3-49 3-105.1 52-125.4c29.4-12.2 61.4-8.7 86.7 6.7c13.1-21.8 32.5-39.4 55.8-50.3L303.6 13.4c-.9-5.5-4.6-10.1-9.7-12.2S282.9-.2 278.5 3L208.2 53.3 137.9 3zM291.4 415.9c-2.5 .6-5.1 .6-7.6-.1c1.4 0 2.8 .1 4.2 .1h3.5zm-46.3-260c-17.3-12.2-40.4-15.6-61.5-6.9c-32.7 13.5-48.2 51-34.6 83.6c3.6 8.7 8.9 16.1 15.3 22.2c10.6-39.7 39.9-71.8 77.8-86.4c.8-4.3 1.8-8.4 3-12.5zM277.4 420c-11-7.4-25.9-4.4-33.3 6.7l-32 48c-7.4 11-4.4 25.9 6.7 33.3s25.9 4.4 33.3-6.7l32-48c7.4-11 4.4-25.9-6.7-33.3zm96 0c-11-7.4-25.9-4.4-33.3 6.7l-32 48c-7.4 11-4.4 25.9 6.7 33.3s25.9 4.4 33.3-6.7l32-48c7.4-11 4.4-25.9-6.7-33.3zm96 0c-11-7.4-25.9-4.4-33.3 6.7l-32 48c-7.4 11-4.4 25.9 6.7 33.3s25.9 4.4 33.3-6.7l32-48c7.4-11 4.4-25.9-6.7-33.3zm96 0c-11-7.4-25.9-4.4-33.3 6.7l-32 48c-7.4 11-4.4 25.9 6.7 33.3s25.9 4.4 33.3-6.7l32-48c7.4-11 4.4-25.9-6.7-33.3zm74.5-116.1c0-39.3-28.4-72.1-65.8-78.7c1.2-5.6 1.9-11.3 1.9-17.2c0-44.2-35.8-80-80-80c-17 0-32.8 5.3-45.8 14.4C433.3 114.6 402.8 96 368 96c-53 0-96 43-96 96l0 1.3c-45.4 7.6-80 47.1-80 94.6c0 53 43 96 96 96H559.9c44.2 0 80-35.8 80-80z"
            fill="currentColor"
          />
        </svg>
        <span
          class="capitalize text-xm md:text-xl font-bold"
        >
          wedrop
        </span>
      </a>
      <nav
        class="pb-6 flex flex-col gap-6 bg-steel-blue z-[10] absolute left-0 w-full h-screen transition-all duration-500 ease-in-out md:order-last md:flex-row md:items-center md:pb-0 md:static md:z-auto md:w-auto md:h-auto md:pt-0 md:pl-9 top-[-890px] pt-auto"
      >
        <a
          aria-current="page"
          class="text-white font-bold text-xl px-6 w-full md:w-auto md:px-1.5 md:mx-1 active:border-b-2 active:border-b-white md:hover:scale-110 active"
          href="/"
        >
          Home
        </a>
        <a
          class="text-white font-bold text-xl px-6 w-full md:w-auto md:px-1.5 md:mx-1 active:border-b-2 active:border-b-white md:hover:scale-110"
          href="/contact"
        >
          Contact
        </a>
      </nav>
      <div
        class="input-group relative flex w-2/5 flex-wrap items-stretch shadow-md z-[100]"
      >
        <input
          aria-describedby="button-addon2"
          aria-label="Search"
          class="form-control relative flex-auto w-4/5 block px-1 py-0.1 text-steel-blue text-bold text-xs font-normal bg-white bg-clip-padding border-white border-2 rounded-l-full transition ease-in-out focus:text-steel-blue focus:bg-white focus:border-white focus:outline-none md:w-auto md:text-xl md:focus:text-xl md:pl-2.5"
          name="search"
          placeholder="Search City..."
          type="search"
        />
        <button
          class="btn inline-block bg-steel-blue w-1/5 md:w-auto text-white font-medium text-xs leading-tight uppercase rounded-r shadow-md border-white border-2 hover:shadow-lg focus:bg-steel-blue  focus:shadow-lg focus:outline-none focus:ring-0 active:shadow-lg active:text-steel-blue active:text-bold transition duration-150 ease-in-out flex items-center justify-center md:px-2 md:py-0.5"
          id="button-addon2"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-magnifying-glass-location text-xs hover:scale-125 px-1.5 py-1.5 md:text-xl md:p-auto"
            data-icon="magnifying-glass-location"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM288 176c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 48.8 46.5 111.6 68.6 138.6c6 7.3 16.8 7.3 22.7 0c22.1-27 68.6-89.8 68.6-138.6zm-48 0c0 17.7-14.3 32-32 32s-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="w-[20px] flex items-center md:hidden z-[100]"
      >
        <button
          class="flex item-center"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars text-xl text-white"
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            id="mobileMenu"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
